### ISSUE TITLE
Error reported in log during sync task, sync doesn't complete all tasks

### ISSUE THREAD

**Issue by notrom** [metabase/metabase#10038]
- Your browser and the version: Firefox 66.0.4 (64-bit)
- Your operating system: Windows 7
- Your databases: MS SQL
- Metabase version: 0.32.7
- Metabase hosting environment: jar file on Windows
- Metabase internal database: Postgres

On every database sync I get this error which appears to be stopping the rest of the sync process from completing:
```
May 24 13:00:11 ERROR metabase.sync.util :: Error running sync step: Multiple methods in multimethod 'fingerprinter' match dispatch value: [:type/DateTime :type/FK] -> [:type/DateTime :type/*] and [:type/* :type/FK], and neither is preferred
("clojure.lang.MultiFn.findAndCacheBestMethod(MultiFn.java:179)"
 "clojure.lang.MultiFn.getMethod(MultiFn.java:150)"
 "clojure.lang.MultiFn.getFn(MultiFn.java:154)"
 "clojure.lang.MultiFn.invoke(MultiFn.java:229)"
 "--> sync.analyze.fingerprint.fingerprinters$fingerprint_fields$iter__39936__39940$fn__39941$fn__39942.invoke(fingerprinters.clj:212)"
 "sync.analyze.fingerprint.fingerprinters$fingerprint_fields$iter__39936__39940$fn__39941.invoke(fingerprinters.clj:211)"
 "sync.analyze.fingerprint.fingerprinters$fingerprint_fields.invokeStatic(fingerprinters.clj:211)"
 "sync.analyze.fingerprint.fingerprinters$fingerprint_fields.invoke(fingerprinters.clj:208)"
 "sync.analyze.fingerprint$fn__45375$fingerprint_table_BANG___45380$fn__45381.invoke(fingerprint.clj:44)"
 "sync.analyze.fingerprint$fn__45375$fingerprint_table_BANG___45380.invoke(fingerprint.clj:40)"
 "sync.analyze.fingerprint$fn__45548$fingerprint_fields_BANG___45553$fn__45554.invoke(fingerprint.clj:161)"
 "sync.analyze.fingerprint$fn__45548$fingerprint_fields_BANG___45553.invoke(fingerprint.clj:157)"
 "sync.analyze.fingerprint$fn__45574$fingerprint_fields_for_db_BANG___45579$fn__45580$fn__45581$iter__45582__45586$fn__45587$fn__45588.invoke(fingerprint.clj:171)"
 "sync.analyze.fingerprint$fn__45574$fingerprint_fields_for_db_BANG___45579$fn__45580$fn__45581$iter__45582__45586$fn__45587.invoke(fingerprint.clj:170)"\
 "sync.analyze.fingerprint$fn__45574$fingerprint_fields_for_db_BANG___45579$fn__45580$fn__45581.invoke(fingerprint.clj:170)"
 "util.date$call_with_effective_timezone.invokeStatic(date.clj:88)"
 "util.date$call_with_effective_timezone.invoke(date.clj:77)"
 "sync.analyze.fingerprint$fn__45574$fingerprint_fields_for_db_BANG___45579$fn__45580.invoke(fingerprint.clj:169)"
 "sync.analyze.fingerprint$fn__45574$fingerprint_fields_for_db_BANG___45579.invoke(fingerprint.clj:164)"
 "sync.analyze$make_analyze_steps$fn__45720.invoke(analyze.clj:110)"
 "sync.util$fn__37755$run_step_with_metadata__37760$fn__37764$fn__37766.invoke(util.clj:356)"
 "sync.util$with_start_and_finish_logging_SINGLEQUOTE_.invokeStatic(util.clj:104)"
 "sync.util$with_start_and_finish_logging_SINGLEQUOTE_.invoke(util.clj:98)"
 "sync.util$with_start_and_finish_debug_logging.invokeStatic(util.clj:121)"
 "sync.util$with_start_and_finish_debug_logging.invoke(util.clj:118)"
 "sync.util$fn__37755$run_step_with_metadata__37760$fn__37764.invoke(util.clj:353)"
 "sync.util$fn__37755$run_step_with_metadata__37760.invoke(util.clj:348)"
 "sync.util$fn__37947$run_sync_operation__37952$fn__37953$fn__37954.invoke(util.clj:441)"
 "sync.util$fn__37947$run_sync_operation__37952$fn__37953.invoke(util.clj:441)"
 "sync.util$fn__37947$run_sync_operation__37952.invoke(util.clj:435)"
 "sync.analyze$fn__45732$analyze_db_BANG___45737$fn__45738$fn__45739.invoke(analyze.clj:127)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:149)"
 "sync.util$do_with_error_handling.invoke(util.clj:144)"
 "sync.util$do_with_error_handling.invokeStatic(util.clj:147)"
 "sync.util$do_with_error_handling.invoke(util.clj:144)"
 "driver$fn__18010.invokeStatic(driver.clj:646)"
 "driver$fn__18010.invoke(driver.clj:646)"
 "sync.util$sync_in_context$fn__37661.invoke(util.clj:140)"
 "sync.util$with_db_logging_disabled$fn__37658.invoke(util.clj:131)"
 "sync.util$with_start_and_finish_logging_SINGLEQUOTE_.invokeStatic(util.clj:104)"
 "sync.util$with_start_and_finish_logging_SINGLEQUOTE_.invoke(util.clj:98)"
 "sync.util$with_start_and_finish_logging$fn__37647.invoke(util.clj:116)"
 "sync.util$with_sync_events$fn__37642.invoke(util.clj:90)"
 "sync.util$with_duplicate_ops_prevented$fn__37633.invoke(util.clj:69)"
 "sync.util$do_sync_operation.invokeStatic(util.clj:168)"
 "sync.util$do_sync_operation.invoke(util.clj:165)"
 "sync.analyze$fn__45732$analyze_db_BANG___45737$fn__45738.invoke(analyze.clj:124)"
 "sync.analyze$fn__45732$analyze_db_BANG___45737.invoke(analyze.clj:119)"
 "task.sync_databases.SyncAndAnalyzeDatabase.execute(sync_databases.clj:42)")
```

The log references multiple fingerprint methods for `[:type/DateTime :type/FK]` , I do have a field in model that is a datetime and a primary key, I use dates in other tables as a foreign key. I suspect this is causing the error.

This error seems to be stopping the rest of the sync process for that database as it  doesn't complete the same tasks when re-scanning as it does for other databases. Below DB ID 2 (Orange) gets the error, DB ID 3 (Green) does not get the error, you'll see that DB 3 has 10 tasks, DB 2 only has 6 and it stops before the "fingerprint-fields" task.
![image](https://user-images.githubusercontent.com/4504437/58297681-a85f7600-7e2c-11e9-9da6-113a4edbd191.png)

Could this be the reason I'm not getting binning on numbers and lat/long as described in issue #7782?

Thanks

**Comment by notrom** [comment]
I've been digging a little further and found that the number fields that do not have binning available do not appear to have a "fingerprint", or  in the Metabase application db. Again not sure if the two are related but there could be something there.

I'd love to have binning working as expected, especially for lat/long values (i.e. grid maps).

**Comment by notrom** [comment]
I manually inserted a reasonable fingerprint value into the Metabase database for a couple of these Lat/long fields and I now get binning in the query builder.

It indicates that the reported error during the schema sync operation is stopping fingerprinting from happening, preventing us from getting a large chunk of functionality from our Metabase setup.

**Comment by sbelak** [comment]
Fixed in #10621
